driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -143,6 +188,7 @@ private void initializeChannel(final AsyncCompletionHandler<Void> handler, final
             bootstrap.handler(new ChannelInitializer<SocketChannel>() {
                 @Override
                 public void initChannel(final SocketChannel ch) {
+                    final ChannelPipeline pipeline = ch.pipeline();
```

Comment:
Nit: No need for local final variables.

Reply:
We need to use a reference to the channel's pipeline in multiple places in this method. While calling `ch.pipeline()` each time we need it is correct, it is an unnecessary call, because we know that the method returns the same reference. We may make some assumptions by looking at the source code, but we should treat it as subject for change, and something that we may believe is "free" for now, may not be free in the future.

Additionally, not calling a method multiple times reduces the number of usages that IDE finds when we want to find usages of the method. This simplifies code navigation.

As for the `final` modifier on local variables, it's a simple way to tell the reader that the variable is assigned once and only once.

Reply:
This codebase has been created by many authors, it looks homogenous because all the developers have tread lightly on the codebase and kept the code style consistent.  Using `final` for local variables isn't a style that has been adopted in the past.

However, with the base line now being java 8 and the need for local final variable in closures perhaps this is a style we should adopt. It is more explicit as you say.

Reply:
I'm happy to consider final local variables as a standard going forward, but currently it's our standard not to, so let's stick with that for now.

Reply:
Also, note that IntellJ now differentiates between mutated and effectively final local variables by underlining the former.

Reply:
I removed the `final` modifier. Is there a document with the style specified (not all style aspects are checked automatically)? I apologize for making not compliant changes, but I have not seen the document that specifies it. If there is none, I'll be picking up the style as I go.